laa_crime_application_store_app/services/v1/application_service.py

```diff
@@ -168,7 +177,12 @@ def update_existing_application(
 
     @staticmethod
     def __get_application_by_id(db: Session, app_id: UUID):
-        return db.query(Application).filter(Application.id == app_id).first()
+        return (
+            db.query(Application)
+            .filter(Application.id == app_id)
+            .with_for_update()
```

Comment:
can you explain that this is added for row locking? And what error do we get if someone tries to edit at the same time and how is it captured and dealt with?

Reply:
There should be a test for race condition...